plugins/user_quota/server/quota.py

```diff
@@ -174,63 +174,55 @@ def _validatePolicy(self, policy):
                            dictionary as if the JSON was already decoded.
         :returns: a validate policy dictionary.
         """
-        validKeys = []
-        for key in dir(self):
-            if key.startswith('_validate_'):
-                validKeys.append(key.split('_validate_', 1)[1])
-        for key in list(policy):
-            if key.startswith('_'):
-                del policy[key]
+        validKeys = [k[10:] for k in dir(self) if k.startswith('_validate_')]
+        policy = {k: v for k, v in six.iteritems(policy) if not k.startswith('_')}
+
         for key in policy:
             if key not in validKeys:
                 raise RestException(
                     '%s is not a valid quota policy key.  Valid keys are %s.' %
                     (key, ', '.join(sorted(validKeys))))
-            funcName = '_validate_' + key
-            policy[key] = getattr(self, funcName)(policy[key])
+            policy[key] = getattr(self, '_validate_' + key)(policy[key])
         return policy
 
-    @access.public
+    @access.user
     @autoDescribeRoute(
         Description('Get quota and assetstore policies for the collection.')
-        .modelParam('id', 'The collection ID', model=Collection, level=AccessType.READ)
+        .modelParam('id', 'The collection ID', model=Collection, level=AccessType.ADMIN)
         .errorResponse('ID was invalid.')
-        .errorResponse('Read permission denied on the collection.', 403)
     )
     def getCollectionQuota(self, collection):
         if QUOTA_FIELD not in collection:
             collection[QUOTA_FIELD] = {}
-        collection[QUOTA_FIELD][
-            '_currentFileSizeQuota'] = self._getFileSizeQuota('collection', collection)
+        collection[QUOTA_FIELD]['_currentFileSizeQuota'] = self._getFileSizeQuota(
+            'collection', collection)
         return self._filter('collection', collection)
 
-    @access.public
+    @access.admin
```

Comment:
The UI allows collection admins (not just site admins) to set collection quotas: https://github.com/girder/girder/blob/master/plugins/user_quota/web_client/templates/collectionViewPoliciesMenu.pug#L1 . This should be made consistent with the API.

Likewise for users: https://github.com/girder/girder/blob/master/plugins/user_quota/web_client/templates/userViewPoliciesMenu.pug#L1

Reply:
That's actually correct behavior in the widgets -- users can *see* their own usage and quota, but not edit it; likewise for collection owners.